From 8ad29832fd05df685039f8c3f4e12fa47ce96bf6 Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Wed, 18 Jul 2012 18:34:50 +0200
Subject: [PATCH 01/18] [PORT FROM MAIN] Add device definitions in HAL for JB

BZ: 47599

Add BT chip state, HAC device selection and set_fmrx_mode
definition.

Change-Id: I4e6a6a5486eaf29d26d2005bde64e9d3bd162d80
Origin-Change-Id: I895fa91dff0fc567f5cd588aad51a1890522b045
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
Signed-off-by: Renaud Viry <renaudx.viry@intel.com>
Reviewed-on: http://android.intel.com:8080/75569
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/hardware/audio.h |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/include/hardware/audio.h b/include/hardware/audio.h
index 3a0962e..573a78d 100644
--- a/include/hardware/audio.h
+++ b/include/hardware/audio.h
@@ -78,6 +78,11 @@ __BEGIN_DECLS
  *  audio device parameters
  */
 
+/* BT chip state */
+#define AUDIO_PARAMETER_KEY_BLUETOOTH_STATE "bluetooth_enabled"
+#define AUDIO_PARAMETER_VALUE_BLUETOOTH_STATE_ON "true"
+#define AUDIO_PARAMETER_VALUE_BLUETOOTH_STATE_OFF "false"
+
 /* BT SCO Noise Reduction + Echo Cancellation parameters */
 #define AUDIO_PARAMETER_KEY_BT_NREC "bt_headset_nrec"
 #define AUDIO_PARAMETER_VALUE_ON "on"
@@ -90,6 +95,11 @@ __BEGIN_DECLS
 #define AUDIO_PARAMETER_VALUE_TTY_HCO "tty_hco"
 #define AUDIO_PARAMETER_VALUE_TTY_FULL "tty_full"
 
+/* HAC device selection */
+#define AUDIO_PARAMETER_KEY_HAC_SETTING "HACSetting"
+#define AUDIO_PARAMETER_VALUE_HAC_ON "ON"
+#define AUDIO_PARAMETER_VALUE_HAC_OFF "OFF"
+
 /* A2DP sink address set by framework */
 #define AUDIO_PARAMETER_A2DP_SINK_ADDRESS "a2dp_sink_address"
 
@@ -371,6 +381,8 @@ struct audio_hw_device {
      */
     int (*set_mode)(struct audio_hw_device *dev, audio_mode_t mode);
 
+    int (*set_fmrx_mode)(struct audio_hw_device *dev, int mode);
+
     /* mic mute */
     int (*set_mic_mute)(struct audio_hw_device *dev, bool state);
     int (*get_mic_mute)(const struct audio_hw_device *dev, bool *state);
-- 
1.7.9

